core/src/components/datetime/utils/state.ts

```diff
@@ -89,13 +89,15 @@ export const isDayDisabled = (
 export const getCalendarDayState = (
   locale: string,
   refParts: DatetimeParts,
-  activeParts: DatetimeParts,
+  activeParts: DatetimeParts | DatetimeParts[],
   todayParts: DatetimeParts,
   minParts?: DatetimeParts,
   maxParts?: DatetimeParts,
   dayValues?: number[]
 ) => {
-  const isActive = isSameDay(refParts, activeParts);
+  const activePartsArray = Array.isArray(activeParts) ? activeParts : [activeParts];
```

Comment:
Why do we not take the recursive approach here like we do in https://github.com/ionic-team/ionic-framework/pull/25514/files/6d336632b022f40c9bd38629bc5e990247f46c06#diff-bc71320a199f4545788c57d1cb7c033c228a9ed44195afa3848d6b88f0a963abR64-R66?

Reply:
Can we add some comments here to clarify what this code is doing? It wasn't immediately clear when I reviewed, but your explanation makes sense.